leftwm-core/src/handlers/window_handler.rs

```diff
@@ -315,16 +311,8 @@ fn set_relative_floating(window: &mut Window, ws: &Workspace, outer: Xyhw) {
     let xyhw = window.requested.map_or_else(
         || ws.center_halfed(),
         |mut requested| {
-            if ws.xyhw.contains_xyhw(&requested) {
-                requested
-            } else {
-                requested.center_relative(outer, window.border);
-                if ws.xyhw.contains_xyhw(&requested) {
-                    requested
-                } else {
-                    ws.center_halfed()
-                }
-            }
```

Comment:
Basically this part, except you can do `requested.center_relative(ws.xyhw, window.border)` instead of `ws.center_halfed()`

Reply:
I'll close this due to the latest commit.